care/facility/api/serializers/patient_consultation.py

```diff
@@ -385,6 +374,60 @@ def validate(self, attrs):
         return validated
 
 
+class PatientConsultationDischargeSerializer(serializers.ModelSerializer):
+    discharge_reason = serializers.ChoiceField(
+        choices=DISCHARGE_REASON_CHOICES, required=True
+    )
+    discharge_notes = serializers.CharField(required=True)
+
+    discharge_date = serializers.DateTimeField(required=False)
+    discharge_prescription = serializers.JSONField(required=False)
+    discharge_prn_prescription = serializers.JSONField(required=False)
+
+    death_datetime = serializers.DateTimeField(required=False, allow_null=True)
+    death_confirmed_doctor = serializers.CharField(required=False, allow_null=True)
+
+    class Meta:
+        model = PatientConsultation
+        fields = (
+            "discharge_reason",
+            "discharge_notes",
+            "discharge_date",
+            "discharge_prescription",
+            "discharge_prn_prescription",
+            "death_datetime",
+            "death_confirmed_doctor",
+        )
+
+    def validate(self, attrs):
+        print(attrs)
+        if attrs.get("discharge_reason") == "EXP":
+            if not attrs.get("death_datetime"):
+                raise ValidationError({"death_datetime": "This field is required"})
+            if not attrs.get("death_confirmed_doctor"):
+                raise ValidationError(
+                    {"death_confirmed_doctor": "This field is required"}
+                )
+            attrs["discharge_date"] = now()
+        elif not attrs.get("discharge_date"):
+            raise ValidationError({"discharge_date": "This field is required"})
+        return attrs
+
+    def save(self, **kwargs):
+        super().save(**kwargs)
```

Comment:
Can we make this a single DB transaction?

Reply:
we have ATOMIC_REQUESTS enabled so do really need it?

Reply:
we have ATOMIC_REQUESTS enabled so do really need it?

Reply:
Let's state it explicitly, especially when we have requests that could produce a 400, Django only considers unhandled exceptions to close transactions to my knowledge.